Sources/TuistGenerator/Generator/KnownAssetTagsGenerator.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+import PathKit
+import TuistGraph
+
+private struct ContentJson: Decodable {
+    struct ContentProperties: Decodable {
+        enum CodingKeys: String, CodingKey {
+            case onDemandResourceTags = "on-demand-resource-tags"
+        }
+
+        let onDemandResourceTags: [String]
+    }
+
+    let properties: ContentProperties
+}
+
+protocol KnownAssetTagsGenerating: AnyObject {
+    func generate(project: Project) throws -> [String]
+}
+
+final class KnownAssetTagsGenerator: KnownAssetTagsGenerating {
```

Comment:
The name of this utility is misleading. Since the term "generate/generator" is reserved for generating Xcode projects and any derived files, I'd not try to use it for anything else. If we understand the logic here correctly, we are fetching the tags from `Content.json` files. What about renaming this utility to `KnownAsseteTagsFetcher`?